### PR TITLE
Craftable Gun Pack mod revamp.

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
I gave this mod a little update to make more "modern" with the experimental. The changes are:

-Added craftable proprietary magazines to smgs.
Internal magazines on an smg?! Fixed it for ya!

-Added 2 accessories slots to all guns.
Even the pipe rifles have them so I added them to these guns.

-Changed burst to 5.
All magazine capacities are multiples of 5, this makes ammo consumption consistent.

-Changes to smg craft requirements:
 *mechanics skill from 6 -> 5. (You are making a makeshift gun, not  a survivor weapon).
 *marksmanship skill from 4 -> 2. (Same as above.)
 *added spring as a component requirement. (Seems obvious that a full auto system uses springs.

-Added .223 pipe rifle.
Already had .308 and .3006 pipe rifles, this was missing.

PS: The big deletion of bootleg ammo here is because of the latest migration ammo. I used the latest experimental's Craft_Gunpowder/cgp_recipes.json but it still would show it as a deletion. I tested it and that part works the same as the  latest experimental's version of the mod.

### CODE REVIEW

File: data/mods/Craft_Gunpowder/cgp_magazine.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    "material": "steel",
+    "symbol": "#",
+    "color": "ltgray",
+    "ammo_type": "22",
+    "capacity": 30,
+    "reliability": 5,
+    "reload_time": 200
+  },
+  {
+    "id": "smg_38_mag",
```
Comment (mugling) [inline data/mods/Craft_Gunpowder/cgp_magazine.json]: Try `"copy-from": "smg_22_mag"`

File: data/mods/Craft_Gunpowder/cgp_ranged.json [hunk 2]
```diff
@@ -179,19 +213,22 @@
     "sight_dispersion": 90,
     "recoil": 0,
     "durability": 4,
-    "burst": 3,
-    "clip_size": 25,
+    "burst": 5,
+    "magazine_well": 1,
+    "magazines": [
```
Comment (mugling) [inline data/mods/Craft_Gunpowder/cgp_ranged.json]: JSON style - if there is only one entry it should go on a single line

File: data/mods/Craft_Gunpowder/cgp_recipes.json [hunk 3]
```diff
@@ -371,5 +337,145 @@
...
+  "category": "CC_AMMO",
+  "subcategory": "CSC_AMMO_OTHER",
+  "skill_used": "fabrication",
+  "skills_required": [ "gun", 1 ],
+  "difficulty": 2,
+  "time": 20000,
+  "reversible": false,
+  "autolearn": true,
+  "qualities":[
+    {"id":"SAW_M","level":1,"amount":1},
```
Comment (mugling) [inline data/mods/Craft_Gunpowder/cgp_recipes.json]: amount: 1 is redundant

File: data/mods/Craft_Gunpowder/cgp_recipes.json [hunk 4]
```diff
@@ -371,5 +337,145 @@
...
+    {"id":"HAMMER","level":1,"amount":1},
+    {"id":"SCREW_FINE","level":1,"amount":1}
+  ],
+  "components": [
+    [
```
Comment (mugling) [inline data/mods/Craft_Gunpowder/cgp_recipes.json]: Again could be compacted further

File: data/mods/Craft_Gunpowder/cgp_magazine.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  {
+    "id": "smg_45_mag",
+    "copy-from": "smg_22_mag",
+    "type": "MAGAZINE",
+    "name": "pipe SMG: .45 magazine",
+    "description": "An improvised magazine made for the pipe SMG: .45, holds 20 rounds.",
+    "ammo_type": "45",
+    "capacity": 20
+  }
+]
```
Comment (mugling) [inline data/mods/Craft_Gunpowder/cgp_magazine.json]: MINOR: No new line

File: data/mods/Craft_Gunpowder/cgp_ranged.json [hunk 6]
```diff
@@ -24,7 +24,7 @@
     "clip_size": 1,
     "reload": 250,
     "loudness" : 150,
-    "valid_mod_locations": [[ "muzzle", 1 ],[ "stock", 1 ]]
+    "valid_mod_locations": [ [ "accessories", 2 ], [ "muzzle", 1 ], [ "stock", 1 ] ]
   },
   {
     "id": "rifle_38",
```
Comment (mugling) [inline data/mods/Craft_Gunpowder/cgp_ranged.json]: Again you should consider `copy-from` (can be a separate PR). The mod becomes more robust the lest duplicate/redundant entries it contains

**Comment by mugling** [comment]
What editor are you using. It's adding the wrong line feeds which necessitates pushing a fix after each of your branches

**Comment by Noctifer-de-Mortem** [comment]
@mugling 
> What editor are you using. It's adding the wrong line feeds which necessitates pushing a fix after each of your branches

I am using notepad++
